plugins/obs-ffmpeg/obs-ffmpeg-mux.c

```diff
@@ -380,6 +380,20 @@ static bool ffmpeg_mux_start(void *data)
 	struct ffmpeg_muxer *stream = data;
 	obs_data_t *settings;
 	const char *path;
+	const char *ext;
+
+	settings = obs_output_get_settings(stream->output);
+	path = obs_data_get_string(settings, "path");
+	ext = strrchr(path, '.');
+	obs_data_release(settings);
+
+	if (strcmp(ext, ".m3u8") == 0) {
+		obs_encoder_t *vencoder;
+		vencoder = obs_output_get_video_encoder(stream->output);
+		settings = obs_encoder_get_settings(vencoder);
+		obs_data_set_bool(settings, "repeat_headers", true);
```

Comment:
change the name of this setting (ex: amf_repeat_headers) and put everything behind a check of the encoder id (h264_texture_amf or h265_texture_amf) (don't bother with the fallback encoders).